<?xml version="1.0"?>
<robot name="roboat" xmlns:xacro="http://roboat.org">
  <xacro:property name="PI" value="3.1415926535897931" />

  <link name="lidar_link"></link>

  <joint name="lidar_link_joint" type="fixed">
    <parent link="lidar_link"/>
    <child link="chassis_link" />
    <origin xyz="0 0 0" rpy="0 0 0" />
  </joint>

  <link name="chassis_link"></link>
  
  <link name="imu_link"> </link>
  <joint name="imu_joint" type="fixed">
    <parent link="chassis_link" />
    <child link="imu_link" />
    <origin xyz="0 -0.1 0" rpy="${PI} 0 ${PI}" />
  </joint>

  <link name="imu_enu_link"> </link>
  <joint name="imu_enu_joint" type="fixed">
    <parent link="imu_link" />
    <child link="imu_enu_link" />
    <origin xyz="0 0 0" rpy="${PI} 0 ${PI}" />
  </joint>

  <link name="velodyne"> </link>
  <joint name="velodyne_joint" type="fixed">
    <parent link="chassis_link" />
    <child link="velodyne" />
    <origin xyz="0 0 0" rpy="0 0 0" />
  </joint>

  <link name="camera_link"> </link>
  <joint name="realsense_joint" type="fixed">
    <parent link="lidar_link" />
    <child link="camera_link" />
    <origin xyz="-0.0504986 0.0332512 -0.0397877" rpy="0 0 0" />
  </joint>

<!-- ? 2023.8.17 使用工具标定的mit数据集的红外-LiDAR 外参 TODO:rpy?
  The calibrated transformation from lidar to camera is:
Transformation matrix: 
-0.0289376  -0.999323 -0.0227257  0.0308632
-0.0156181  0.0231845  -0.999609 -0.0413318
  0.999459 -0.0285714 -0.0162785  0.0507736
         0          0          0          1
Qaternion(qx qy qz qw): -0.490957  0.516817 -0.497362 -0.494461
Translation(tx ty tz):   0.0308632 -0.0413318  0.0507736
CalibrationResult:      The calibrated transformation from camera to lidar is:
Transformation matrix: 
-0.0289376 -0.0156181   0.999459 -0.0504986
 -0.999323  0.0231845 -0.0285714  0.0332512
-0.0227257  -0.999609 -0.0162785 -0.0397877
         0          0         -0          1
Qaternion(qx qy qz qw): -0.490957  0.516817 -0.497362  0.494461
Translation(tx ty tz):  -0.0504986  0.0332512 -0.0397877 -->

  <link name="navsat_link"> </link>
  <joint name="navsat_joint" type="fixed">
    <parent link="chassis_link" />
    <child link="navsat_link" />
    <origin xyz="-0.2 0 0.2" rpy="0 0 0" />
  </joint>

  <link name="loam_camera"> </link>
  <joint name="_loam_camera_joint" type="fixed">
    <parent link="chassis_link" />
    <child link="loam_camera" />
    <origin xyz="0 0 0" rpy="1.570796 0 1.570796" />
  </joint>

</robot>
